### PR TITLE
Adding Esri 2020 Land Cover/Land Use Dataset

### PR DESCRIPTION
**Pull request by nilsleh**
This PR adds the [Esri 2020 Land Cover/Land Use Dataset](https://www.arcgis.com/home/item.html?id=d6642f8a4f6d4685a24ae2dc0c73d4ac). It contains global labels for 10 different classes from Sentinel 2 imagery at 10m resolution. The methodology is described in the following [paper](https://ieeexplore.ieee.org/abstract/document/9553499). Individual tiles can have 30000x28000 pixel resolution. 

Here are a couple of examples:
esri2020B
esri2020A

### CODE REVIEW

File: torchgeo/datasets/esri2020.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+from .geo import RasterDataset
+from .utils import download_url, extract_archive
+
+
+class Esri2020(RasterDataset, abc.ABC):
+    """Esri 2020 Land Cover Dataset.
+
+    The 'Esri 2020 Land Cover dataset
+    <https://www.arcgis.com/home/item.html?id=fc92d38533d440078f17678ebc20e8e2>'
```
Comment (adamjstewart) [inline torchgeo/datasets/esri2020.py:21]:
```suggestion
    The `Esri 2020 Land Cover dataset
    <https://www.arcgis.com/home/item.html?id=fc92d38533d440078f17678ebc20e8e2>`_
```

File: torchgeo/datasets/esri2020.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    2 Water
+    3 Trees
+    4 Grass
+    5 Flooded Vegetation
+    6 Crops
+    7 Scrub/Shrub
+    8 Built Area
+    9 Bare Ground
+    10 Snow/Ice
+    11 Clouds
```
Comment (adamjstewart) [inline torchgeo/datasets/esri2020.py:38]:
```suggestion
    The 10 classes are:
    
    1. No Data
    2. Water
    3. Trees
    4. Grass
    5. Flooded Vegetation
    6. Crops
    7. Scrub/Shrub
    8. Built Area
    9. Bare Ground
    10. Snow/Ice
    11. Clouds
```

File: torchgeo/datasets/esri2020.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+    A more detailed explanation of the invidual classes can be found
+    'here <https://www.arcgis.com/home/item.html?id=fc92d38533d440078f17678ebc20e8e2>'
```
Comment (adamjstewart) [inline torchgeo/datasets/esri2020.py:41]:
```suggestion
    `here <https://www.arcgis.com/home/item.html?id=fc92d38533d440078f17678ebc20e8e2>`_
```

File: torchgeo/datasets/esri2020.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+
+    If you use this dataset please cite the following paper:
+
+    * <https://ieeexplore.ieee.org/document/9553499>
```
Comment (adamjstewart) [inline torchgeo/datasets/esri2020.py:45]:
```suggestion
    * https://ieeexplore.ieee.org/document/9553499
```

File: torchgeo/datasets/esri2020.py (hunk 2 above)
Comment (adamjstewart) [reply torchgeo/datasets/esri2020.py:38]: This is 11 classes, not 10. We should either start counting from 0 or remove "No Data"

File: torchgeo/datasets/esri2020.py [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset.
+
+        Raises:
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+        """
+        # Check if the extracted file already exists
+        pathname = os.path.join(self.root, "**", self.filename_glob)
+        if len(glob.glob(pathname)) != 0:
```
Comment (adamjstewart) [inline torchgeo/datasets/esri2020.py:110]:
```suggestion
        if glob.glob(pathname):
```

File: torchgeo/datasets/__init__.py [hunk 6]
```diff
@@ -120,6 +121,7 @@
     "CV4AKenyaCropType",
     "DFC2022",
     "EnviroAtlas",
+    "Esri2020",
```
Comment (adamjstewart) [inline torchgeo/datasets/__init__.py:124]: I would move this into the GeoDataset section above

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: torchgeo/datasets/esri2020.py [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+    4. Flooded Vegetation
+    5. Crops
+    6. Scrub/Shrub
+    7. Built Area
+    8. Bare Ground
+    9. Snow/Ice
+    10. Clouds
+
+    A more detailed explanation of the invidual classes can be found
+    `here <https://www.arcgis.com/home/item.html?id=fc92d38533d440078f17678ebc20e8e2>`_
```
Comment (adamjstewart) [inline torchgeo/datasets/esri2020.py:42]:
```suggestion
    `here <https://www.arcgis.com/home/item.html?id=fc92d38533d440078f17678ebc20e8e2>`_.
```

**Review by adamjstewart — APPROVED** [review]

**Comment by daviddao** [comment]
@nilsleh this is amazing!! Thank you!

**Pull request merged**